filesystem/squashfs_superblock.ksy

```diff
@@ -0,0 +1,109 @@
+meta:
+  id: squashfs_superblock
+  title: SquashFS superblock
+  file-extension:
+    - squashfs
+    - snap
+  endian: le
+  license: CC-0
+  xref:
+    wikidata: Q389314
+doc-ref: https://github.com/AgentD/squashfs-tools-ng/blob/master/doc/format.txt
```

Comment:
It's a good habit to describe the format usage field in a few sentences in the top-level `doc` as well. The idea is that people wondering what the "SquashFS superblock" is don't have to go to Wikipedia, as they get some basic idea already from the [format gallery](https://formats.kaitai.io/) page.

Reply:
For me the usage was to fix timestamps in the SquashFS image - https://github.com/yakshaveinc/linux/issues/39#issuecomment-667940076 - and I decided to try making a superblock first. Coming back to the rest of the format when I get more time.

Reply:
@abitrolly Well, I didn't really meant what you intend to do with the parser, but rather what the "SquashFS" is at all. Is it a filesystem? Where can it be found in practice, on CDs, hard drives or somewhere else? Is it a proprietary format? Which systems / applications can work with it?

Reply:
@generalmimon added the doc to the top. Feel free to fix mistakes if any.